test/exprtest/blas1_iface_test.cpp

```diff
@@ -0,0 +1,156 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas1_iface_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas1_test.hpp"
+
+typedef ::testing::Types<blas1_test_args<double>> BlasTypes;
+
+TYPED_TEST_CASE(BLAS1_Test, BlasTypes);
+
+REGISTER_SIZE(::RANDOM_SIZE, interface1_test)
+REGISTER_STRD(1, interface1_test)
+REGISTER_PREC(float, 1e-4, interface1_test)
+REGISTER_PREC(double, 1e-6, interface1_test)
+
+TYPED_TEST(BLAS1_Test, interface1_test) {
+  using ScalarT = typename TypeParam::scalar_t;
+  using ExecutorType = typename TypeParam::executor_t;
+  using TestClass = BLAS1_Test<TypeParam>;
+  using test = class interface1_test;
+
+  size_t size = TestClass::template test_size<test>();
+  size_t strd = TestClass::template test_strd<test>();
+  ScalarT prec = TestClass::template test_prec<test>();
+
+  std::cout << "size == " << size << std::endl;
```

Comment:
Protect all stdout with a macro so it can be disabled easily. Alternatively use some function that allows to enable or disable a verbose mode, but at this point simply if/def cout with a VERBOSE macro